improver_tests/categorical/decision_tree/test_ApplyDecisionTree.py

```diff
@@ -46,9 +48,62 @@
     find_threshold_coordinate,
     get_threshold_coord_name_from_probability_name,
 )
-from improver.synthetic_data.set_up_test_cubes import set_up_probability_cube
+from improver.synthetic_data.set_up_test_cubes import (
+    set_up_probability_cube,
+    set_up_variable_cube,
+)
 
-from . import wxcode_decision_tree
+from . import deterministic_diagnostic_tree, wxcode_decision_tree
+
+
+@pytest.fixture()
+def precip_cube() -> Cube:
+    """
+    Set up a precipitation rate, deterministic cube.
+    """
+
+    data = np.full((7, 3), dtype=np.float32, fill_value=5)
+    cube = set_up_variable_cube(
+        data,
+        name="precipitation_rate",
+        units="mm/s",
+        time=dt(2017, 10, 10, 12, 0),
+        frt=dt(2017, 10, 10, 11, 0),
+    )
+    return cube
+
+
+@pytest.fixture()
+def hail_cube() -> Cube:
+    """
+    Set up a hail rate, deterministic cube.
+    """
+
+    data = np.full((7, 3), dtype=np.float32, fill_value=0)
+    cube = set_up_variable_cube(
+        data,
+        name="hail_rate",
+        units="mm/s",
+        time=dt(2017, 10, 10, 12, 0),
+        frt=dt(2017, 10, 10, 11, 0),
+    )
+    return cube
+
+
+@pytest.mark.parametrize(
+    "precip_fill,hail_fill,expected", ((1, 1, 2), (1, 0, 1), (0, 0, 0))
```

Comment:
You could add `(0, 1, 0)` to show that if hail disagrees with precip, it is ignored.

Reply:
Done